File: e2e/cypress-javascript/support/api/ApiClient.js
const testConfig = require('@/test.config');

class ApiClient {
  constructor() {
    this.baseUrl = testConfig.services.api.url;
  }

  createTask(task) {
    return cy.request('POST', `${this.baseUrl}/tasks`, task).its('body');
  }

  deleteTask(id) {
    return cy.request('DELETE', `${this.baseUrl}/tasks/${id}`);
  }

  getTasks() {
    return cy.request('GET', `${this.baseUrl}/tasks`).its('body');
  }
}

module.exports = { ApiClient };

File: e2e/capybara-ruby/support/api/api_client.rb
# frozen_string_literal: true

require 'faraday'

require_relative '../test_config'

class ApiClient
  def initialize
    @base_url = TestConfig.api_url
    @connection = Faraday.new(url: @base_url) do |faraday|
      faraday.request :json
      faraday.response :json
      faraday.adapter Faraday.default_adapter
    end
  end

  def create_task(task)
    response = @connection.post('tasks', task)
    response.body
  end

  def delete_task(id)
    @connection.delete("tasks/#{id}")
  end

  def get_tasks
    response = @connection.get('tasks')
    response.body
  end
end
